### ISSUE TITLE
Understanding the evaluation report at 25 Epoch

### ISSUE THREAD

**Issue by AlbertoMCS** [tensorpack/tensorpack#1271]
+ If you did something with tensorpack and it failed, PLEASE STOP HERE and use the "Unexpected Problems /
  Bugs" category.

+ Your question is probably answered in [tutorials](http://tensorpack.readthedocs.io/tutorial/index.html#user-tutorials). Read it first.

+ We answer "HOW to do X with Tensorpack" for a well-defined specific X.
  X must be something that you conceptually know how to do, but are unable to do due to lack of knowledge about Tensorpack.
  We don't answer vague questions like "How to do semantic segmentation in Tensorpack".

+ We also answer "HOW/WHY Tensorpack does X" for some X that Tensorpack or its examples are doing.

Some typical questions that we DO NOT answer:

+ "Could you improve/implement an example/paper?":

  We have no plans to do so. We don't consider feature
  requests for examples or implement a paper for you.
  If you don't know how to do something yourself, you may ask a usage question.

+ "The examples do not perform as expected after I change the models/dataset/parameters/etc.":

  Tensorpack maintainers make sure the examples perform well without modifications.
  But it's your job to pick the model and parameters that are suitable for your own situation.
  We cannot help with such questions unless they appear to be a bug in tensorpack.

+ "Why my own model doesn't perform well?", "I don't understand this paper you implement.",
  "How should I change the parameters for my own dataset?":

  We do not answer machine learning questions.

You can also use gitter (https://gitter.im/tensorpack/users) for more casual discussions.

Hi, 

I trained Faster-RCNN with Mapillary dataset (18.000 trainset, 2.000 valset). At the epoch 25 a evaluation report is generated showing some values and ratios about precision/average precision etc. I would like to understand exactly how my model is performing according to this report. I checked tensorpack tutorials but I didn't find anything about this. Could anybody shed some light about how to read into these figures (I expect the model to be low accurate as I just went to 25 epochs). 

Below the report at 25 epoch
Thanks
Alberto.

loading annotations into memory...
Done (t=2.60s)
creating index...
index created!
[0712 15:10:00 @dataset.py:46] Instances loaded from /home/federicolondon2019/tensorpack/COCO/DIR/annotations/instances_val2017.json.
Loading and preparing results...
DONE (t=0.51s)
creating index...
index created!
Running per image evaluation...
Evaluate annotation type *bbox*
DONE (t=22.44s).
Accumulating evaluation results...
DONE (t=1.42s).
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.077
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.152
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.070
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.009
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.087
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.206
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.067
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.121
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.124
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.009
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.130
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.318
Running per image evaluation...
Evaluate annotation type *segm*
DONE (t=26.99s).
Accumulating evaluation results...
DONE (t=1.41s).
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.062
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.126
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.056
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.006
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.066
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.169
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.054
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.097
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.100
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.006
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.103
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.264
[0712 15:10:54 @monitor.py:467] GPUUtil/0: 30.461
[0712 15:10:54 @monitor.py:467] GPUUtil/1: 30.901
[0712 15:10:54 @monitor.py:467] PeakMemory(MB)/gpu:0: 7146.8
[0712 15:10:54 @monitor.py:467] QueueInput/queue_size: 8.2718e-25
[0712 15:10:54 @monitor.py:467] fastrcnn_losses/box_loss: 0.19387
[0712 15:10:54 @monitor.py:467] fastrcnn_losses/label_loss: 0.3364
[0712 15:10:54 @monitor.py:467] fastrcnn_losses/label_metrics/accuracy: 0.88264
[0712 15:10:54 @monitor.py:467] fastrcnn_losses/label_metrics/false_negative: 0.27507
[0712 15:10:54 @monitor.py:467] fastrcnn_losses/label_metrics/fg_accuracy: 0.64587
[0712 15:10:54 @monitor.py:467] fastrcnn_losses/num_fg_label: 121
[0712 15:10:54 @monitor.py:467] learning_rate: 0.01
[0712 15:10:54 @monitor.py:467] maskrcnn_loss/accuracy: 0.86461
[0712 15:10:54 @monitor.py:467] maskrcnn_loss/fg_pixel_ratio: 0.51308
[0712 15:10:54 @monitor.py:467] maskrcnn_loss/maskrcnn_loss: 0.3055
[0712 15:10:54 @monitor.py:467] maskrcnn_loss/pos_accuracy: 0.44743
[0712 15:10:54 @monitor.py:467] multilevel_roi_align/fpn_map_rois_to_levels/num_roi_level2: 332.23
[0712 15:10:54 @monitor.py:467] multilevel_roi_align/fpn_map_rois_to_levels/num_roi_level3: 129.38
[0712 15:10:54 @monitor.py:467] multilevel_roi_align/fpn_map_rois_to_levels/num_roi_level4: 47.433
[0712 15:10:54 @monitor.py:467] multilevel_roi_align/fpn_map_rois_to_levels/num_roi_level5: 2.9581
[0712 15:10:54 @monitor.py:467] multilevel_roi_align_mask/fpn_map_rois_to_levels/num_roi_level2: 101.5
[0712 15:10:54 @monitor.py:467] multilevel_roi_align_mask/fpn_map_rois_to_levels/num_roi_level3: 15.319
[0712 15:10:54 @monitor.py:467] multilevel_roi_align_mask/fpn_map_rois_to_levels/num_roi_level4: 4.0567
[0712 15:10:54 @monitor.py:467] multilevel_roi_align_mask/fpn_map_rois_to_levels/num_roi_level5: 0.12451
[0712 15:10:54 @monitor.py:467] rpn_losses/box_loss: 0.18759
[0712 15:10:54 @monitor.py:467] rpn_losses/label_loss: 0.088851
[0712 15:10:54 @monitor.py:467] rpn_losses/level2/box_loss: 0.14784
[0712 15:10:54 @monitor.py:467] rpn_losses/level2/label_loss: 0.066923
[0712 15:10:54 @monitor.py:467] rpn_losses/level2/label_metrics/precision_th0.1: 0.84085
[0712 15:10:54 @monitor.py:467] rpn_losses/level2/label_metrics/precision_th0.2: 0.87109
[0712 15:10:54 @monitor.py:467] rpn_losses/level2/label_metrics/precision_th0.5: 0.9165
[0712 15:10:54 @monitor.py:467] rpn_losses/level2/label_metrics/recall_th0.1: 0.98252
[0712 15:10:54 @monitor.py:467] rpn_losses/level2/label_metrics/recall_th0.2: 0.97967
[0712 15:10:54 @monitor.py:467] rpn_losses/level2/label_metrics/recall_th0.5: 0.95424
[0712 15:10:54 @monitor.py:467] rpn_losses/level2/num_pos_anchor: 79.087
[0712 15:10:54 @monitor.py:467] rpn_losses/level2/num_valid_anchor: 193.89
[0712 15:10:54 @monitor.py:467] rpn_losses/level3/box_loss: 0.023504
[0712 15:10:54 @monitor.py:467] rpn_losses/level3/label_loss: 0.01348
[0712 15:10:54 @monitor.py:467] rpn_losses/level3/label_metrics/precision_th0.1: 0.8107
[0712 15:10:54 @monitor.py:467] rpn_losses/level3/label_metrics/precision_th0.2: 0.8401
[0712 15:10:54 @monitor.py:467] rpn_losses/level3/label_metrics/precision_th0.5: 0.90322
[0712 15:10:54 @monitor.py:467] rpn_losses/level3/label_metrics/recall_th0.1: 0.97413
[0712 15:10:54 @monitor.py:467] rpn_losses/level3/label_metrics/recall_th0.2: 0.9689
[0712 15:10:54 @monitor.py:467] rpn_losses/level3/label_metrics/recall_th0.5: 0.95427[0712 15:10:54 @monitor.py:467] rpn_losses/level3/num_pos_anchor: 18.342[0712 15:10:54 @monitor.py:467] rpn_losses/level3/num_valid_anchor: 43.247[0712 15:10:54 @monitor.py:467] rpn_losses/level4/box_loss: 0.012254[0712 15:10:54 @monitor.py:467] rpn_losses/level4/label_loss: 0.0067451[0712 15:10:54 @monitor.py:467] rpn_losses/level4/label_metrics/precision_th0.1: 0.81179[0712 15:10:54 @monitor.py:467] rpn_losses/level4/label_metrics/precision_th0.2: 0.84432[0712 15:10:54 @monitor.py:467] rpn_losses/level4/label_metrics/precision_th0.5: 0.87846[0712 15:10:54 @monitor.py:467] rpn_losses/level4/label_metrics/recall_th0.1: 0.93437[0712 15:10:54 @monitor.py:467] rpn_losses/level4/label_metrics/recall_th0.2: 0.93294[0712 15:10:54 @monitor.py:467] rpn_losses/level4/label_metrics/recall_th0.5: 0.9197[0712 15:10:54 @monitor.py:467] rpn_losses/level4/num_pos_anchor: 9.5476[0712 15:10:54 @monitor.py:467] rpn_losses/level4/num_valid_anchor: 14.535[0712 15:10:54 @monitor.py:467] rpn_losses/level5/box_loss: 0.0039125[0712 15:10:54 @monitor.py:467] rpn_losses/level5/label_loss: 0.0016214[0712 15:10:54 @monitor.py:467] rpn_losses/level5/label_metrics/precision_th0.1: 0.75506[0712 15:10:54 @monitor.py:467] rpn_losses/level5/label_metrics/precision_th0.2: 0.77868[0712 15:10:54 @monitor.py:467] rpn_losses/level5/label_metrics/precision_th0.5: 0.83692[0712 15:10:54 @monitor.py:467] rpn_losses/level5/label_metrics/recall_th0.1: 0.84224[0712 15:10:54 @monitor.py:467] rpn_losses/level5/label_metrics/recall_th0.2: 0.84216[0712 15:10:54 @monitor.py:467] rpn_losses/level5/label_metrics/recall_th0.5: 0.82995[0712 15:10:54 @monitor.py:467] rpn_losses/level5/num_pos_anchor: 3.3331[0712 15:10:54 @monitor.py:467] rpn_losses/level5/num_valid_anchor: 4.2694[0712 15:10:54 @monitor.py:467] rpn_losses/level6/box_loss: 7.1645e-05[0712 15:10:54 @monitor.py:467] rpn_losses/level6/label_loss: 8.2145e-05[0712 15:10:54 @monitor.py:467] rpn_losses/level6/label_metrics/precision_th0.1: 0.50754[0712 15:10:54 @monitor.py:467] rpn_losses/level6/label_metrics/precision_th0.2: 0.5077[0712 15:10:54 @monitor.py:467] rpn_losses/level6/label_metrics/precision_th0.5: 0.51404[0712 15:10:54 @monitor.py:467] rpn_losses/level6/label_metrics/recall_th0.1: 0.51822[0712 15:10:54 @monitor.py:467] rpn_losses/level6/label_metrics/recall_th0.2: 0.51822[0712 15:10:54 @monitor.py:467] rpn_losses/level6/label_metrics/recall_th0.5: 0.51793[0712 15:10:54 @monitor.py:467] rpn_losses/level6/num_pos_anchor: 0.038035[0712 15:10:54 @monitor.py:467] rpn_losses/level6/num_valid_anchor: 0.063274[0712 15:10:54 @monitor.py:467] sample_fast_rcnn_targets/num_bg: 391[0712 15:10:54 @monitor.py:467] sample_fast_rcnn_targets/num_fg: 121[0712 15:10:54 @monitor.py:467] sample_fast_rcnn_targets/proposal_metrics/best_iou_per_gt: 0.42607[0712 15:10:54 @monitor.py:467] sample_fast_rcnn_targets/proposal_metrics/recall_iou0.3: 0.56637[0712 15:10:54 @monitor.py:467] sample_fast_rcnn_targets/proposal_metrics/recall_iou0.5: 0.45517[0712 15:10:54 @monitor.py:467] total_cost: 1.7057
[0712 15:10:54 @monitor.py:467] val2017-mAP(bbox)/IoU=0.5: 0.15158[0712 15:10:54 @monitor.py:467] val2017-mAP(bbox)/IoU=0.5:0.95: 0.077494[0712 15:10:54 @monitor.py:467] val2017-mAP(bbox)/IoU=0.75: 0.070238[0712 15:10:54 @monitor.py:467] val2017-mAP(bbox)/large: 0.20624[0712 15:10:54 @monitor.py:467] val2017-mAP(bbox)/medium: 0.087207[0712 15:10:54 @monitor.py:467] val2017-mAP(bbox)/small: 0.0092415[0712 15:10:54 @monitor.py:467] val2017-mAP(segm)/IoU=0.5: 0.12563[0712 15:10:54 @monitor.py:467] val2017-mAP(segm)/IoU=0.5:0.95: 0.061988[0712 15:10:54 @monitor.py:467] val2017-mAP(segm)/IoU=0.75: 0.055521[0712 15:10:54 @monitor.py:467] val2017-mAP(segm)/large: 0.1692[0712 15:10:54 @monitor.py:467] val2017-mAP(segm)/medium: 0.066399[0712 15:10:54 @monitor.py:467] val2017-mAP(segm)/small: 0.0060503[0712 15:10:54 @monitor.py:467] wd_cost: 0.59354[0712 15:10:54 @group.py:48] Callbacks took 1276.102 sec in total. EvalCallback: 21 minutes

**Comment by ppwwyyxx** [comment]
These are coco metrics and are unrelated to tensorpack.
See http://cocodataset.org/#detection-eval
You may find more details about them in earlier papers introducing the PASCAL VOC dataset, or some online blog posts.

**Comment by Adblu** [comment]
@ppwwyyxx Could you point to those papers ? Cannot find it anywhere. I want to understand all those metrics.
